types/node-jose/index.d.ts

```diff
@@ -0,0 +1,346 @@
+
+        toPEM(isPrivate?: boolean): string;
+        toJSON(isPrivate?: boolean, excluded?: string[]): object;
+        thumbprint(hash?: string): Promise<string>;
+    }
+}
+
+export namespace JWS {
+    function createSign(keys: JWK.Key | JWK.Key[]): Signer;
+    function createSign(
+        options: {
+            format?: 'compact' | 'flattened';
+            alg?: string;
+            compact?: boolean;
+            fields?: object;
+        },
+        key: JWK.Key | JWK.Key[]
+    ): Signer;
+
+    /**
+     * Using a keystore.
+     */
+    function createVerify(
+        input?: string | JWK.Key | JWK.KeyStore | object,
+        opts?: { allowEmbeddedKey?: boolean; algorithms?: string[]; handlers?: any }
+    ): Verifier;
+
+    interface CreateSignResult {
+        signResult: object;
+    }
+
+    interface Signer {
+        update(input: Buffer | string, encoding?: string): this;
+        final(): Promise<CreateSignResult>;
+    }
+
+    interface BaseResult {
+        /**
+         * the combined 'protected' and 'unprotected' header members
+         */
+        header: object;
+        /**
+         * the signed content
+         */
+        payload: Buffer;
+        /**
+         * The key used to verify the signature
+         */
+        key: JWK.Key;
+        protected: string[];
+    }
+
+    interface VerificationResult extends BaseResult {
+        /**
+         * the verified signature
+         */
+        signature: Buffer | string;
+    }
+
+    interface Verifier {
+        verify(input: string, opts?: { allowEmbeddedKey?: boolean }): Promise<VerificationResult>;
+    }
+
+    interface Exp {
+        complete(jws: any): any;
+    }
+
+    interface VerifyOptions {
+        allowEmbeddedKey?: boolean;
+        algorithms?: string[];
+        handlers: { exp: boolean | Exp };
+    }
+}
+
+interface ParseReturn {
+    type: 'JWS' | 'JWE';
+    format: 'compact' | 'json';
+    input: Buffer | string | object;
+    header: object;
+    perform: (ks: JWK.KeyStore) => Promise<JWE.DecryptResult> | Promise<JWS.VerificationResult>;
+}
+
+export function parse(input: Buffer | string | object): ParseReturn;
+
+export namespace parse {
+    function compact(input: Buffer | string | object): ParseReturn;
+
+    function json(input: Buffer | string | object): ParseReturn;
+}
+
+export namespace util {
+    function asBuffer(input: string | Buffer, encoding?: string): Buffer;
+
+    function randomBytes(len: number): Buffer;
+
+    namespace base64url {
+        function decode(base64url: string): string;
+
+        function encode(buffer: string | Buffer, encoding?: string): string;
+    }
+
+    namespace utf8 {
+        function decode(input: string): string;
+
+        function encode(input: string): string;
+    }
+}
+
+declare const _default: {
+    JWA: typeof JWA;
+    JWE: typeof JWE;
+    JWS: typeof JWS;
+    JWK: typeof JWK;
+    parse: typeof parse;
+    util: typeof util;
+};
+export default _default;
```

Comment:
Looking at the source, it doesn't seem like the module actually has a default export https://github.com/cisco/node-jose/blob/master/lib/index.js so you can remove this section. All the necessary members are already exported above using `export` keywords.

Reply:
hi @minestarks ,
I fixed it.
can you please verify this.